src/main/java/io/vertx/redis/client/CachingRedisClient.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2019 Red Hat, Inc.
+ * <p>
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Apache License v2.0 which accompanies this distribution.
+ * <p>
+ * The Eclipse Public License is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ * <p>
+ * The Apache License v2.0 is available at
+ * http://www.opensource.org/licenses/apache2.0.php
+ * <p>
+ * You may elect to redistribute this code under either of these licenses.
+ */
+package io.vertx.redis.client;
+
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import io.vertx.redis.client.impl.CachingRedisClientImpl;
+import io.vertx.redis.client.impl.cache.CacheKey;
+
+import java.util.Collection;
+
+/**
+ * A {@link Redis} client wrapper that implements client-side caching.
+ *
+ * @see <a href="https://redis.io/docs/manual/client-side-caching/">Client-side caching in Redis</a>
+ */
+public interface CachingRedisClient extends Redis {
```

Comment:
should be named `CachingRedis` for consistency matters

Reply:
Done.